packages/dto/src/comment.rs

```diff
@@ -1,34 +1,43 @@
 #![allow(unused_variables, unused)]
-use crate::Result;
+use crate::{Result, User};
 #[cfg(feature = "server")]
 use by_axum::aide;
 use by_macros::api_model;
 use by_types::QueryResponse;
 use validator::ValidationError;
 
-#[api_model(base = "/v1/topics/:topic-id/comments", table = comments, iter_type = QueryResponse)]
+#[api_model(base = "/v1/topics/:topic-id/comments", table = comments, action_by_id = like, iter_type = QueryResponse)]
 pub struct Comment {
     #[api_model(summary, primary_key, read_action = [get_comment, find_by_id])]
     pub id: i64,
     #[api_model(summary, auto = insert)]
     pub created_at: u64,
 
-    #[api_model(summary)]
-    pub profile_url: String,
+    #[api_model(many_to_one = topics)]
+    pub topic_id: i64,
+    #[api_model(many_to_one = users)]
+    pub author_id: i64,
 
     // TODO: custom function from topics and users joined table
     // #[api_model(summary)]
     // pub choice: Option<String>,
-    #[api_model(summary)]
-    pub nickname: String,
     #[api_model(summary, action = comment, related = String)]
     pub content: String,
 
-    #[api_model(many_to_one = topics)]
-    pub topic_id: i64,
-
-    #[api_model(summary, one_to_many = user_comments, aggregator = count)]
+    #[api_model(summary, one_to_many = comments_likes, foreign_key = comment_id, aggregator = count)]
     pub likes: u64,
-    #[api_model(many_to_many = user_comments, foreign_table_name = users, foreign_primary_key = user_id, foreign_reference_key = comment_id, aggregator = exist, exist)]
+    #[api_model(many_to_many = comments_likes, foreign_table_name = users, foreign_primary_key = user_id, foreign_reference_key = comment_id, aggregator = exist, exist, unique)]
     pub is_liked: bool,
 }
+
+#[api_model(base = "/v1/topics/:topic-id/comments", table = comment_likes, iter_type = QueryResponse)]
+pub struct CommentLike {
+    #[api_model(summary, primary_key, read_action = find_by_id)]
+    pub id: i64,
+    #[api_model(summary, auto = insert)]
+    pub created_at: u64,
+    #[api_model(many_to_one = comments)]
+    pub comment_id: i64,
+    #[api_model(many_to_one = users)]
+    pub user_id: i64,
+}
```

Comment:
정식 프로덕션에서 사용되는 코드라면 아래와 같이 되는게 맞아요. 근데 얘를 따로 뭔가 해야할만큼 필요한게 있어보이지는 않네요? 왜 따로 만들었지? 아무튼 이런식으로 되야 합니다.

```suggestion
#[api_model(base = "/v1/topics/:topic-id/comments", table = user_comments, iter_type = QueryResponse)]
pub struct CommentLike {
    #[api_model(summary, primary_key)]
    pub id: i64,
    #[api_model(summary, auto = insert)]
    pub created_at: u64,
    #[api_model(many_to_one = comments)]
    pub comment_id: i64,
    #[api_model(many_to_one = users)]
    pub user_id: i64,
}
```

Reply:
user_comments 테이블은 이미 따로 있고 유저가 좋아요를 누른 테이블이 따로 필요하지 않을까 싶어서 저렇게 했습니다. 같은 테이블로 취급하면 안되지 않나요? @hackartists